staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -321,8 +321,15 @@ func (f *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 	f.addCall("instance-metadata-by-provider-id")
 	f.addressesMux.Lock()
 	defer f.addressesMux.Unlock()
+
+	providerID := node.Spec.ProviderID
+	_, ok := f.ExtIDErr[types.NodeName(node.Name)]
```

Comment:
This should probably check `ExtID` instead of `ExtIDErr`, and fake cloud provider could likely use a new field like `MetadataErr` to mock errors when calling `InstanceMetadata`

Reply:
I thought if there is `ExtIDErr` associated with the nodeName, that means `ExtID` is an empty map, but if `ExtIDErr` doesn't exist, we have ExtID available.
But you are right, maybe checking `ExtID` is more straightforward
will update!